java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java

```diff
@@ -1409,9 +1409,16 @@ public void setConvertVectorElement(int elementNum) {
         seconds = SerializationUtils.convertFromUtc(local, seconds);
       }
       long wholeSec = (long) Math.floor(seconds);
-      timestampColVector.time[elementNum] = wholeSec * 1000;
-      timestampColVector.nanos[elementNum] =
-          1_000_000 * (int) Math.round((seconds - wholeSec) * 1000);
+
+      if (seconds >= Long.MAX_VALUE || seconds <= Long.MIN_VALUE) { // overflow
```

Comment:
I think this should be seconds * 1000, since otherwise the time value can overflow.

Reply:
agree, let's multiple first and then check